src/main/java/cc/polyfrost/oneconfig/config/Config.java

```diff
@@ -261,6 +263,18 @@ protected final void generateOptionList(Object instance, Class<?> targetClass, O
             } else if (field.isAnnotationPresent(HUD.class)) {
                 HUDUtils.addHudOptions(page, field, instance, this);
             }
+            else if(field.isAnnotationPresent(Category.class)){
+                Category categoryAnnotation = field.getAnnotation(Category.class);
+                String name = categoryAnnotation.name();
+                Object categoryInstance = ConfigUtils.getField(field, instance);
+                if(categoryInstance == null) continue;
+                if(categoryInstance instanceof cc.polyfrost.oneconfig.config.elements.OptionCategory) {
+                    page.categories.put(name, (cc.polyfrost.oneconfig.config.elements.OptionCategory) categoryInstance);
+                }
+                else{
+                    addCategory(categoryInstance,categoryInstance.getClass(),page,mod, name,migrate);
```

Comment:
if `addCategory` is only used here, why not just implement it into here instead of adding a whole new method?

Reply:
i wanted to make the code more understandble, without it alot of nesting would happend